wallet/how-to/use-non-evm-networks/starknet/create-a-simple-starknet-dapp.md

```diff
@@ -322,89 +436,103 @@ src/
 
-Call the `balanceOf` method to fetch the balance of the connected account:
+        // Wait for the transaction to be accepted.
+        await account.waitForTransaction(transferTxHash);
 
-```typescript title="TokenBalanceAndTransfer.tsx"
-const balance = await erc20.balanceOf(walletAddress);
-const formattedBalance = balance / Math.pow(10, 18);
+        // Refresh the balance after the transfer.
+        const newBalance = await erc20.balanceOf(account.address) as bigint;
+        setBalance(Number(newBalance / 10n ** 18n));
+      }
+    } catch (error) {
+      console.error("Error during transfer:", error);
+    }
+  }
 ```
 
-### 4.5. Transfer tokens
+:::note ABI and contract address
 
-To transfer tokens, fill out the `transfer` method call and execute the transaction using the `AccountInterface`.
+The contract address for STRK (an ERC-20 token) on Sepolia testnet is `0x049D36570D4e46f48e99674bd3fcc84644DdD6b96F7C741B1562B82f9e004dC7`.
+You can find the ABI of the ERC-20 contract on the **Code** tab in [Voyager](https://voyager.online/).
 
-Use the following example for reference:
+:::
 
-```typescript title="TokenBalanceAndTransfer.tsx"
-import { Call } from "starknet";
+### 4.6. Update `App.tsx`
 
-// Define the transfer parameters.
-const recipientAddress = "0x78662e7352d062084b0010068b99288486c2d8b914f6e2a55ce945f8792c8b1";
-const amountToTransfer = 1n * 10n ** 18n; // 1 token (assuming 18 decimals).
+Call the `TokenBalanceAndTransfer` component in `App.tsx`.
+Add the followinwg to the header of `App.tsx` to import the component:
 
-const transferCall: Call = erc20.populate("transfer", {
-  recipient: recipientAddress,
-  amount: amountToTransfer,
-});
+```typescript title="App.tsx"
+import { TokenBalanceAndTransfer } from "./components/TokenBalanceAndTransfer";
+```
 
-// Execute the transfer.
-const { transaction_hash: transferTxHash } = await AccountInterface.execute(transferCall);
+Ensure you call the token address in the `TokenBalanceAndTransfer` component.
+Add the following code to `App.tsx`, where the `TokenBalanceAndTransfer` component is called with the token address:
 
-// Wait for the transaction to be accepted on Starknet.
-await AccountInterface.waitForTransaction(transferTxHash);
+```typescript title="App.tsx"
+{walletAccount &&
+    <TokenBalanceAndTransfer account={walletAccount} tokenAddress="0x04718f5a0fc34cc1af16a1cdee98ffb20c31f5cd61d6ab07201858f4287c938d" />
+}
 ```
 
-<div class="row">
-  <div class="column">
-    <img src={require("../../../assets/starknet-tutorial-transfer-token.png").default} alt="Starknet transfer token" width="400" style={{border: "1px solid #DCDCDC"}} />
-  </div>
-</div>
-
-### 4.6. Full example
+### 4.7. Full example
 
-The following a full example of displaying the balance of an ERC-20 token and performing a transfer:
+The following is a full example of displaying the balance of an ERC-20 token and performing a transfer:
 
 <Tabs>
```

Comment:
I am now getting this compilation error for the token transfer part of the tutorial, using the provided full examples. Maybe I'm not importing the component correctly into `App.tsx`? Can you show me your project file? It might be helpful to add `App.tsx` as a tab to the full example here, too.

![Screenshot 2024-10-28 at 11 40 46 AM](https://github.com/user-attachments/assets/0989ef7a-652e-49df-895f-c37eb77618c1)

Reply:
Hi @alexandratran did you update the `tsconfig.json`?

![image](https://github.com/user-attachments/assets/00bd8467-83d3-4533-a48d-a84f23adb91e)

Reply:
I'll push a commit to add the `App.tsx` file as well